pkg/core/mempool/mempool.go

```diff
@@ -103,6 +108,20 @@ func NewMempool(eventBus *eventbus.EventBus, rpcBus *rpcbus.RPCBus, verifier tra
 
 	acceptedBlockChan, _ := consensus.InitAcceptedBlockUpdate(eventBus)
 
+	// Enable rate limiter from config
+	fromConfig := config.Get().Mempool.PropagateTimeout
+
+	var limiter *rate.Limiter
+
+	if len(fromConfig) > 0 {
+		timeout, err := time.ParseDuration(config.Get().Mempool.PropagateTimeout)
+		if err != nil {
+			log.WithError(err).Fatal("could not parse mempool propagation timeout")
+		}
+
+		limiter = rate.NewLimiter(rate.Every(timeout), 1)
```

Comment:
Burst of 1 every 100ms sounds quite low. Maybe consider a higher value (2 or 3 without message priority, higher with priority)

Reply:
I'll add it. Could be good to play with it on benchmark tests too.